### ISSUE TITLE
ANGLE is not work with some Graphic Card(e.g. Intel HD Graphics Family)

### ISSUE THREAD

**Issue by andy65007** [nspire-emus/firebird#82]
![2017-05-07_184716](https://cloud.githubusercontent.com/assets/8511187/25780257/2e4ddab8-3357-11e7-9a6b-ce47132b0209.png)
![2017-05-07_185216](https://cloud.githubusercontent.com/assets/8511187/25780259/3344ddfa-3357-11e7-8f96-3147848d2736.png)
we can simply add one line code: "QCoreApplication::setAttribute(Qt::AA_UseDesktopOpenGL);" in main.c to avoid this problem.
this solution only fix

more detail please refer to:  Strange Behaviour with a Multi-Monitor-Set and OpenGL(https://forum.qt.io/topic/69396/strange-behaviour-with-a-multi-monitor-set-and-opengl/14)

**Comment by Vogtinator** [comment]
It looks to me like it already tries OpenGL 2.0, so desktop opengl by default. Are all the listed dlls available in the same directory as firebird.exe?

**Comment by andy65007** [comment]
PS: the real error msg is not point ANGLE error when  run  firebird.exe. it says:" api-ms-win-core-libraryloader-l1-2-0.dll is missing"
![api-ms-win-core-libraryloader-l1-2-0-dll-is-missing-from-your-computer](https://cloud.githubusercontent.com/assets/8511187/25780356/0a9a545a-3359-11e7-8154-8159ee4f6cab.jpg)

**Comment by Vogtinator** [comment]
That's not good, that dll is part of windows itself. I'm afraid that your installation is apparently broken.

**Comment by andy65007** [comment]
I try firebird on several pc, none of them have that dll,

**Comment by andy65007** [comment]
I think maybe it's a bug of ANGLE

**Comment by Vogtinator** [comment]
I don't think so, can you add the .dll manually to the firebird directory?

**Comment by andy65007** [comment]
yes, I have tried that, manually add dll "api-ms-win-core-libraryloader-l1-2-0.dll" to APP folder, It does not work. so, I think this issue is not about operation system broken or dll missing, also not about app installation broken, just add "QCoreApplication::setAttribute(Qt::AA_UseDesktopOpenGL);" to main.c can solve this problem.

**Comment by Vogtinator** [comment]
> It does not work.

What happens?

>  just add "QCoreApplication::setAttribute(Qt::AA_UseDesktopOpenGL);" to main.c can solve this problem.

That is not a solution, it will break on platforms which do not have HW OpenGL.

**Comment by andy65007** [comment]
1. I really dont know 'what hapens', this issue takes me several days to find out 'what happens' till I find this article: "Strange Behaviour with a Multi-Monitor-Set and OpenGL"(which I mentioned earlier, it solves my problem).
2. Obviously,  add '"QCoreApplication::setAttribute(Qt::AA_UseDesktopOpenGL);" will bring cross-platform problems. And I already mentioned this at first thread in this issue, hope some one can use a better way to solve it, because I'm not good at Qt programming.

**Comment by andy65007** [comment]
The model of my laptop is lenovo E420 with two graphic cards, hope you can find a similer one for testing or just close this issue becasue it rarely happen. We keep this issue here just for someone like me, it help them out.
